### PR TITLE
Db unit integration

### PR DESCRIPTION
**Pull request by sebastian-mikucki**
dbUnit integration with sample test, ready to usage

### CODE REVIEW

File: src/test/java/com/guysfromusa/carsgame/repositories/BaseRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import org.springframework.test.context.support.DirtiesContextTestExecutionListener;
+import org.springframework.test.context.transaction.TransactionalTestExecutionListener;
+
+/**
+ * Created by Sebastian Mikucki, 26.02.18
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = SpringContextConfiguration.class)
+@TestExecutionListeners({DependencyInjectionTestExecutionListener.class,
+        DirtiesContextTestExecutionListener.class,
```
Comment (tbradlo) [inline src/test/java/com/guysfromusa/carsgame/repositories/BaseRepositoryTest.java]: do we need DirtiesContext executor? hopefully we wont need to use DirtiesContext

File: src/test/java/com/guysfromusa/carsgame/repositories/MapRepositoryTest.java [hunk 2]
```diff
@@ -14,20 +11,22 @@
...
-@RunWith(SpringJUnit4ClassRunner.class)
-@ContextConfiguration(classes = SpringContextConfiguration.class)
-public class MapRepositoryTest {
+public class MapRepositoryTest extends BaseRepositoryTest {
 
     @Inject
     private MapRepository mapRepository;
 
     @Test
+    @DatabaseSetup("/insert-map-data.xml")
```
Comment (tbradlo) [inline src/test/java/com/guysfromusa/carsgame/repositories/MapRepositoryTest.java]: @all please use one xml file per one test mehtod. then xmls are smaller and easier to understand.

File: src/test/resources/insert-map-data.xml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<?xml version='1.0' encoding='UTF-8'?>
```
Comment (tbradlo) [inline src/test/resources/insert-map-data.xml]: would be good to have DTD here

Comment (sebastian-mikucki) [reply src/test/resources/insert-map-data.xml]: yes, in progress

**Pull request merged**